platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java

```diff
@@ -938,6 +946,10 @@ private boolean isAppInFeatureList(Feature feature, String applicationName) {
   public void serviceChanged(ServiceEvent serviceEvent) {
     if (serviceEvent.getType() == ServiceEvent.REGISTERED) {
       BundleContext context = getContext();
+      if (context == null) {
```

Comment:
I am realizing that there are a lot of checks for the bundle context being null. I think it might be cleaner to remove these checks and instead throw an IllegalStateException in the getContext() method instead of returning null. There are several files in this pr that would benefit from this change.